src/chainstate/stacks/index/storage.rs

```diff
@@ -672,7 +1240,12 @@ pub struct TrieStorageTransaction<'a, T: MarfTrieId>(TrieStorageConnection<'a, T
 pub struct TrieStorageConnection<'a, T: MarfTrieId> {
     pub db_path: &'a str,
     db: SqliteConnection<'a>,
+    blobs: Option<&'a mut TrieFile>,
     data: &'a mut TrieStorageTransientData<T>,
+    cache: &'a mut TrieCache<T>,
+    bench: &'a mut TrieBenchmark,
+    pub hash_calculation_mode: TrieHashCalculationMode,
+    unconfirmed_block_id: Option<u32>,
```

Comment:
What is `unconfirmed_block_id`?

Reply:
Also, what is its relationship to `data.cur_block_id`?

Reply:
`unconfirmed_block_id` is the row ID of the trie representing unconfirmed state.  This is an odd case, because such tries can be re-loaded from disk, modified, re-written, and dropped (whereas all other tries are write-once-read-many and cannot be dropped).  This field is only set to `Some(..)` if the trie being operated upon is such a trie.

`self.cur_block_id` remains the row ID of the trie that is currently open.  It may or may not be equal to `self.unconfirmed_block_id`.

Reply:
I'll add rustdoc for this.

Reply:
N.B. The reason `self.unconfirmed_block_id` is set here, and not in the constructor, is because it follows the semantics of `self.cur_block_id` -- the row ID isn't resolved until we go to open the block.